src/lib/netlist/plib/ppmf.h

```diff
@@ -153,277 +119,341 @@ namespace plib {
+				// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+				std::uint8_t *vtable_base = *reinterpret_cast<std::uint8_t **>(o_p_delta);
+				// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+				func = *reinterpret_cast<generic_function *>(vtable_base + m_function - 1);
 			}
-			else if (PMFINTERNAL == 3)
-			{
-				const int SINGLE_MEMFUNCPTR_SIZE = sizeof(void (generic_class::*)());
+			object = o_p_delta;
+		}
+
+		// actual state
+		uintptr_t m_function;         // first item can be one of two things:
+													//    if even, it's a pointer to the function
+													//    if odd, it's the byte offset into the vtable
+		ptrdiff_t m_this_delta;       // delta to apply to the 'this' pointer
+	};
+
+	template <>
+	class mfp_raw<PPMF_TYPE_INTERNAL_ARM>
+	{
+	public:
+		// construct from any member function pointer
+		using generic_function = void (*)();
+
+		template<typename MemberFunctionType>
+		mfp_raw(MemberFunctionType mftp)
+		: m_function(0), m_this_delta(0)
+		{
+			static_assert(sizeof(*this) >= sizeof(MemberFunctionType), "size mismatch");
+			*reinterpret_cast<MemberFunctionType *>(this) = mftp; // NOLINT
+		}
 
+	//private:
+		// extract the generic function and adjust the object pointer
+		void convert_to_generic(generic_function &func, mfp_generic_class *&object) const
+		{
+			if ((m_this_delta & 1) == 0)
+			{
 				// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+				object = reinterpret_cast<mfp_generic_class *>(reinterpret_cast<std::uint8_t *>(object) + (m_this_delta >> 1));
+				// NOLINTNEXTLINE(performance-no-int-to-ptr,cppcoreguidelines-pro-type-reinterpret-cast)
 				func = reinterpret_cast<generic_function>(m_function);
-				if (m_size == SINGLE_MEMFUNCPTR_SIZE + sizeof(int))
-					// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
-					object = reinterpret_cast<generic_class *>(reinterpret_cast<std::uint8_t *>(object) + m_this_delta);
+			}
+			else
+			{
+				// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+				object = reinterpret_cast<mfp_generic_class *>(reinterpret_cast<std::uint8_t *>(object));
+
+				// otherwise, it is the byte index into the vtable where the actual function lives
+				// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+				std::uint8_t *vtable_base = *reinterpret_cast<std::uint8_t **>(object);
+				// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+				func = *reinterpret_cast<generic_function *>(vtable_base + m_function + m_this_delta - 1);
 			}
 		}
 
 		// actual state
 		uintptr_t               m_function;         // first item can be one of two things:
 													//    if even, it's a pointer to the function
 													//    if odd, it's the byte offset into the vtable
-		int                     m_this_delta;       // delta to apply to the 'this' pointer
+		ptrdiff_t               m_this_delta;       // delta to apply to the 'this' pointer
```

Comment:
This comment describing the standard Itanium ABI snuck through – it’s not correct for the ARM variant here.